x-pack/test/api_integration/apis/management/cross_cluster_replication/auto_follow_pattern.js

```diff
@@ -9,63 +9,85 @@ import Chance from 'chance';
 import { API_BASE_PATH, REMOTE_CLUSTERS_API_BASE_PATH } from './constants';
 
 const chance = new Chance();
+const CHARS_POOL = 'abcdefghijklmnopqrstuvwxyz';
+const getRandomName = () => chance.string({ pool: CHARS_POOL });
+const CLUSTER_NAME = `test-${getRandomName()}`;
+const BASE_AUTOFOLLOW_PATTERN = '/auto_follow_patterns';
+const AUTOFOLLOW_PATTERNS_API_BASE_PATH = API_BASE_PATH + BASE_AUTOFOLLOW_PATTERN;
 
 export default function ({ getService }) {
+  let remoteClusterAdded = false;
+  let autoFollowPatternsCreated = [];
   const supertest = getService('supertest');
 
-  const CLUSTER_NAME = 'test_cluster';
-  const BASE_AUTOFOLLOW_PATTERN = 'auto_follow_patterns';
-  const CHARS_POOL = 'abcdefghijklmnopqrstuvwxyz';
-
   const addCluster = async (name = CLUSTER_NAME) => {
-    const uri = `${REMOTE_CLUSTERS_API_BASE_PATH}`;
-
-    return await supertest
-      .post(uri)
-      .set('kbn-xsrf', 'xxx')
-      .send({
-        "name": name,
-        "seeds": [
-          "localhost:9300"
-        ],
-        "skipUnavailable": true,
-      });
+    remoteClusterAdded = true;
+
+    return (
+      await supertest
+        .post(`${REMOTE_CLUSTERS_API_BASE_PATH}`)
+        .set('kbn-xsrf', 'xxx')
+        .send({
+          "name": name,
+          "seeds": [
+            "localhost:9300"
+          ],
+          "skipUnavailable": true,
+        })
+    );
   };
 
-  const deleteCluster = async (name = CLUSTER_NAME) => {
-    const uri = `${REMOTE_CLUSTERS_API_BASE_PATH}/${name}`;
-
-    return await supertest
-      .delete(uri)
-      .set('kbn-xsrf', 'xxx')
-      .expect(200);
+  const deleteCluster = (name = CLUSTER_NAME) => {
+    if (!remoteClusterAdded) {
```

Comment:
If we could remove this variable then I think I'd find the code a bit clearer.  If we remove the `expect(200)` from line 48, then does that remove the need for `remoteClusterAdded`? If that isn't possible, then perhaps we could just call `deleteCluster` directly after the specific test suites which require it.